pkg/kubelet/cm/container_manager_linux.go

```diff
@@ -576,6 +609,68 @@ func (cm *containerManagerImpl) setFsCapacity() error {
 	return nil
 }
 
+// TODO: move the GetResources logic to PodContainerManager.
+func (cm *containerManagerImpl) GetResources(pod *v1.Pod, container *v1.Container, activePods []*v1.Pod) (*kubecontainer.RunContainerOptions, error) {
+	opts := &kubecontainer.RunContainerOptions{}
+	// Gets devices, mounts, and envs from device plugin handler.
+	glog.V(3).Infof("Calling device_plugin AllocateDevices")
+	// Maps to detect duplicate settings.
+	devsMap := make(map[string]string)
+	mountsMap := make(map[string]string)
+	envsMap := make(map[string]string)
+	allocResps, err := cm.devicePluginHandler.Allocate(pod, container, activePods)
+	if err != nil {
+		return opts, err
+	}
+	for _, resp := range allocResps {
+		for _, devRuntime := range resp.Spec {
+			for _, dev := range devRuntime.Devices {
+				if d, ok := devsMap[dev.ContainerPath]; ok {
+					glog.V(3).Infof("skip existing device %s %s", dev.ContainerPath, dev.HostPath)
+					if (d != dev.HostPath) {
+						glog.Errorf("Container device %s has conflicting mapping host devices: %s and %s",
```

Comment:
My feeling is this kind of misconfiguration may not be fatal. E.g., say the same device is accessible through multiple paths and either is fine. Since we log error here, I think we can make this information accessible through npm for debugging purpose.